ArbiShop/products/views.py

```diff
@@ -61,3 +54,37 @@ class ProductDetailView(DetailView):
     model = Product
     template_name = 'products/product_detail.html'
     context_object_name = 'product'
+
+
+class CategoryProductsView(FilterView, ListView):
+    """
+    Shows a paginated and filtered list of products in a category.
+
+    Combines FilterView and ListView to display products in a specific category with filtering.
+    Also provides additional context for the current category and all available categories.
+    """
+
+    model = Product
+    template_name = 'products/category_product_listing.html'
+    context_object_name = 'products'
+    paginate_by = 12
```

Comment:
You can keep this in a config table with keys and value so changing pagination can be on the go instead of committing in code and deploying everytime